README.md
# dialog_spinner

A Flutter package to make dialog mode of spinner in android available in flutter

## Installing

* To use this package, add ```dialog_spinner``` as a [dependency in your pubspec.yaml file](https://flutter.dev/docs/development/packages-and-plugins/using-packages)

## Example

    import 'package:flutter/material.dart';
    import 'package:dialog_spinner/dialog_spinner.dart';

    void main() => runApp(MaterialApp(home: MyApp()));

    class MyApp extends StatefulWidget {
      final List<String> spinnerList = ["Item 1", "Item 2","Item 3", "Item 4", "Item 5"];

      @override
      State<StatefulWidget> createState() {
        return _MyAppState();
      }
    }

    class _MyAppState extends State<MyApp> {
      String currentItem;

      @override
      void initState() {
        super.initState();
        currentItem = widget.spinnerList[0];
      }

      @override
      Widget build(BuildContext context) {
        return Scaffold(
          appBar: AppBar(
            title: Text("Dialog Spinner Example"),
          ),
          body: Column(
            children: <Widget>[
              DialogSpinner(
                spinnerList: widget.spinnerList,
                onValueChanged: (int index) {
                  setState(() {
                    currentItem = widget.spinnerList[index];
                  });
                },
              ),
              Expanded(
                child: Center(
                  child: Text(
                    currentItem + " is selected.",
                  ),
                ),
              )
            ],
          ),
        );
      }
    }


CHANGELOG.md
## [0.0.1]

* Initial Release

## [0.1.0]

* Improved spinner styling features

## [0.1.1]

* Added example in recomended format

## [0.1.2]

* Updated README.md and example/README.md

## [0.1.3]

* Updated link in README.md

example/dialog_spinner_example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:dialog_spinner_example/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/dialog_spinner.dart
library dialog_spinner;

import 'package:flutter/material.dart';

class DialogSpinner extends StatefulWidget {
  final List<String> spinnerList;
  final Function onValueChanged;

  final TextAlign textAlign;
  final TextStyle style;
  final TextAlign spinnerTextAlign;
  final TextStyle spinnerStyle;
  final EdgeInsets margin;
  final EdgeInsets spinnerMargin;
  final EdgeInsets padding;
  final EdgeInsets spinnerPadding;
  final Icon iconDropDown;
  final Color color;
  final Decoration dialogDecoration;
  final Decoration spinnerDecoration;

  final double elevation;
  final double width;
  final double height;

  DialogSpinner({
    @required this.spinnerList,
    @required this.onValueChanged(int index),
    this.textAlign = TextAlign.center,
    this.style = const TextStyle(
        fontSize: 20.0, fontWeight: FontWeight.bold, color: Colors.black),
    this.spinnerTextAlign = TextAlign.center,
    this.spinnerStyle = const TextStyle(
        fontSize: 18.0, fontWeight: FontWeight.normal, color: Colors.black),
    this.margin = const EdgeInsets.all(0),
    this.spinnerMargin = const EdgeInsets.all(5),
    this.padding = const EdgeInsets.all(0),
    this.spinnerPadding = const EdgeInsets.all(10.0),
    this.iconDropDown = const Icon(Icons.arrow_drop_down),
    this.color,
    this.dialogDecoration,
    this.spinnerDecoration,
    this.elevation = 2.0,
    this.width,
    this.height,
  });

  @override
  State<StatefulWidget> createState() {
    return _DialogSpinnerState();
  }
}

class _DialogSpinnerState extends State<DialogSpinner> {
  String hint;

  @override
  void initState() {
    super.initState();
    hint = widget.spinnerList[0];
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      width: widget.width,
      height: widget.height,
      margin: widget.margin,
      child: Card(
        elevation: widget.elevation,
        margin: EdgeInsets.all(0),
        child: FlatButton(
          padding: widget.padding,
          materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
          color: widget.color,
          child: Row(
            children: <Widget>[
              Expanded(
                child: Text(
                  hint,
                  textAlign: widget.textAlign,
                  style: widget.style,
                ),
              ),
              widget.iconDropDown,
            ],
          ),
          onPressed: () {
            showDialog(
                context: context,
                builder: (BuildContext context) {
                  return AlertDialog(
                    contentPadding: EdgeInsets.all(0),
                    content: Container(
                      width: MediaQuery.of(context).size.width,
                      height: MediaQuery.of(context).size.height,
                      decoration: widget.dialogDecoration,
                      child: Scrollbar(
                        child: ListView.builder(
                          itemCount: widget.spinnerList.length,
                          itemBuilder: (BuildContext context, int index) {
                            return Card(
                              elevation: widget.elevation,
                              margin: widget.spinnerMargin,
                              child: InkWell(
                                child: Container(
                                  decoration: widget.spinnerDecoration,
                                  padding: widget.spinnerPadding,
                                  child: Text(
                                    widget.spinnerList[index],
                                    textAlign: widget.spinnerTextAlign,
                                    style: widget.spinnerStyle,
                                  ),
                                ),
                                onTap: () {
                                  Navigator.pop(context);
                                  setState(() {
                                    hint = widget.spinnerList[index];
                                  });
                                  widget.onValueChanged(index);
                                },
                              ),
                            );
                          },
                        ),
                      ),
                    ),
                  );
                });
          },
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:dialog_spinner/dialog_spinner.dart';

void main() => runApp(MaterialApp(
      title: "Test App",
      home: MyApp(),
    ));

class MyApp extends StatefulWidget {
  final List<String> spinnerList = [
    "Item 1",
    "Item 2",
    "Item 3",
    "Item 4",
    "Item 5",
    "Item 6",
    "Item 7",
    "Item 8",
    "Item 9",
    "Item 10",
    "Item 11",
    "Item 12",
    "Item 13",
    "Item 14",
    "Item 15",
  ];

  @override
  State<StatefulWidget> createState() {
    return _MyAppState();
  }
}

class _MyAppState extends State<MyApp> {
  String currentItem;

  @override
  void initState() {
    super.initState();
    currentItem = widget.spinnerList[0];
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Test App"),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.start,
        children: <Widget>[
          DialogSpinner(
            spinnerList: widget.spinnerList,
            onValueChanged: (int index) {
              setState(() {
                currentItem = widget.spinnerList[index];
              });
            },
            width: 300,
            height: 50,
            margin: EdgeInsets.all(10.0),
            color: Colors.blue,
            dialogDecoration: BoxDecoration(
              color: Colors.blue,
            ),
            spinnerDecoration: BoxDecoration(
              color: Colors.white,
            ),
            style: TextStyle(
              color: Colors.white,
              fontSize: 20.0,
              fontWeight: FontWeight.bold,
            ),
            spinnerStyle: TextStyle(
              color: Colors.black,
              fontSize: 18.0,
              fontWeight: FontWeight.normal,
            ),
          ),
          Expanded(
            child: Center(
              child: Text(
                currentItem + " is selected.",
                style: TextStyle(
                  fontSize: 20.0,
                  fontWeight: FontWeight.bold,
                  color: Colors.black,
                ),
              ),
            ),
          )
        ],
      ),
    );
  }
}
